pkg/router/server.go

```diff
@@ -0,0 +1,398 @@
+// chromeとかで開いたページはGETとして認識される
+
+package main
+
+import (
+	// "fmt"
+	"log"
+	"net/http"
+
+	// "encoding/json"
+	"github.com/ant0ine/go-json-rest/rest"
+	// "../apisec/apisec.go"
+	"github.com/satackey/procon31server/pkg/apispec"
+
+	"sync"
+)
+
+func main() {
+
+	// var a apisec.Match
+
+	log.Println("localhost:3000 でサーバを起動しました")
+	api := rest.NewApi()
+
+	api.Use(rest.DefaultDevStack...)
+	router, err := rest.MakeRouter(
+		rest.Get("/ping", PingsThings),
+
+		rest.Get("/matches", GetAllMatch),
+
+		// 実際には降ってくるけど, 練習環境ではそれがないのでpostを用意しておく
+		// rest.Post("mathces", PostMatch),
+		// PathExp must start with /
+		rest.Post("/matches", PostMatch),
+
+		// CRUDのC Post or Put
+		rest.Get("/matches/:id", GetUpdateAction),
+		// rest.Delete("/matches/:id", DeleteMatch),
+		rest.Post("/matches/:id/advanceaction", PostAction),
+		// rest.Get("/matches/:id/action"),
+		rest.Post("/matches/:id/action", PostUpdate),
+		// // rest.Post("/matches/:id/action", NewPostUpdate),
+		// rest.Post("/matchesnew/:id", GetFieldStatus),
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+	api.SetApp(router)
+	log.Fatal(http.ListenAndServe(":3000", api.MakeHandler()))
+}
+
+// Ping は
+type Ping struct {
+	Status string `json:"status"`
+}
+
+// PingsThings は
+func PingsThings(w rest.ResponseWriter, r *rest.Request) {
+
+	MyResponse(w, "OK", 202)
+	return
+
+}
+
+// こういう構造体をapisec.goから引っ張ってきたい
+// type Status struct {
+//     Id int `json:id`
+// }
+
+// type Match struct {
+// 	Id             int    `json:"id"`             // 試合のid
+// 	IntervalMillis int    `json:"intervalMillis"` // ターンとターンの間の時間
+// 	MatchTo        string `json:"matchTo"`        // 対戦相手の名前
+// 	TeamID         int    `json:"teamID"`         // 自分のteamid
+// 	TurnMillis     int    `json:"turnMillis"`     // 1ターンの(制限?)時間
+// 	Turns          int    `json:"turns"`          // 試合のターン数
+// }
+
+// var store = map[int]*Match{}
+
+var lock = sync.RWMutex{}
+
+func PostMatch(w rest.ResponseWriter, r *rest.Request) {
+	// country := new(Country)
+	matchstruct := apispec.Match{}
```

Comment:
- 構造体であることは明らかにわかると思うので struct は消しましょう
- ほとんどの場合で構造体はポインタで扱います

```suggestion
	match := &apispec.Match{}
```

Reply:
ポインタで(参照という形で)扱った方が高速だからですか?

Reply:
変数のスコープが同じならどっちでもいいのだけど、どっちでも良いものはポインタに統一したいと思っています